presto-iceberg/src/main/java/com/facebook/presto/iceberg/TypeConverter.java

```diff
@@ -209,16 +227,55 @@ public static org.apache.iceberg.types.Type fromRow(RowType type, int startId)
         return Types.StructType.of(fields);
     }
 
+    public static org.apache.iceberg.types.Type fromRow(
+            RowType type,
+            String columnName,
+            Map<String, Integer> columnNameToIdMapping)
+    {
+        List<Types.NestedField> fields = new ArrayList<>();
+        for (RowType.Field field : type.getFields()) {
+            String name = field.getName().orElseThrow(() ->
+                    new PrestoException(NOT_SUPPORTED, "Row type field does not have a name: " + type.getDisplayName()));
```

Comment:
Does this mean we can't support anonymous row types?

Reply:
Yes, although I haven't changed anything as part of this PR, it has been like this.

Reply:
@imjalpreet : If this error is only for the purpose of converting to Iceberg, then maybe we should change message to "Cannot convert Row type field " + type.getDisplayName() + " to Iceberg"